### PR TITLE
Add `is_auto_command` config check to conditionally publish output an…

### PR DESCRIPTION
**Pull request by mrT23**
### **User description**
…d update GitHub App to set this flag

___

### **PR Type**
enhancement, configuration changes

___

### **Description**
- Added `is_auto_command` configuration setting to control conditional publishing of outputs.
- Updated GitHub App handler to set `is_auto_command` to `True` for new PRs.
- Modified code suggestions, PR description, and PR reviewer tools to check `is_auto_command` before publishing outputs.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>github_app.py</strong><dd><code>Set `is_auto_command` to True in GitHub App handler</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
pr_agent/servers/github_app.py

<li>Added setting <code>is_auto_command</code> to <code>True</code> in the GitHub App handler for <br>new PRs.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/986/files#diff-3fb3f174152732d1b69953c8bd81b8a0a30f0e42100dc626d932da8371851608">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pr_code_suggestions.py</strong><dd><code>Conditional output publishing based on `is_auto_command`</code>&nbsp; </dd></summary>
<hr>
      
pr_agent/tools/pr_code_suggestions.py

- Added check for `is_auto_command` before publishing output.

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/986/files#diff-b57ba775e741d6f80bc4f8154b71330c011dae0ac43f3d0197e785b3e6b7117b">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pr_description.py</strong><dd><code>Conditional PR description publishing based on `is_auto_command`</code></dd></summary>
<hr>
      
pr_agent/tools/pr_description.py

<li>Added check for <code>is_auto_command</code> before publishing PR description.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/986/files#diff-66130451adce278a098a5770e2db52b12050205fcd4d80c13f813615d4449abb">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pr_reviewer.py</strong><dd><code>Conditional review comment publishing based on `is_auto_command`</code></dd></summary>
<hr>
      
pr_agent/tools/pr_reviewer.py

<li>Added check for <code>is_auto_command</code> before publishing review comments.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/986/files#diff-8e265068e189a06852605cc694b03e92b523b4f8162077a2f3455ced4cdad8dc">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Configuration changes
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>configuration.toml</strong><dd><code>Add `is_auto_command` configuration setting</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
pr_agent/settings/configuration.toml

<li>Added <code>is_auto_command</code> configuration setting with default value <code>false</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/986/files#diff-66cfda5143e484ee53ecf7aa0df7dca8ad0b181256f4b0675905db35bcbbae78">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: pr_agent/tools/pr_description.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ async def run(self):
             relevant_configs = {'pr_description': dict(get_settings().pr_description),
                                 'config': dict(get_settings().config)}
             get_logger().debug("Relevant configs", artifacts=relevant_configs)
-            if get_settings().config.publish_output:
+            if get_settings().config.publish_output and not get_settings().config.get('is_auto_command', False):
                 self.git_provider.publish_comment("Preparing PR description...", is_temporary=True)
```
Comment (mrT23) [inline pr_agent/tools/pr_description.py:83]: /ask explain these changes

**Comment by mrT23** [comment]
/improve
--pr_code_suggestions.extra_instructions="
Emphasize the following:
- Does the code logic cover relevant edge cases?
- Is the code logic clear and easy to understand?
- Is the code logic efficient?
- 
"
--pr_code_suggestions.commitable_code_suggestions=false

**Review by hussam789 — APPROVED** [review]

**Pull request merged**